lib/widgets/path_editor/cards/graph_settings_card.dart

```diff
@@ -229,15 +253,16 @@ class _GraphSettingsCardState extends State<GraphSettingsCard> {
       mainAxisSize: MainAxisSize.max,
       children: [
         Checkbox(
-          value: widget.prefs.getBool(GraphEditor.prefShowCurvature),
+          value: _showCurvature,
           activeColor: GraphEditor.colorCurvature,
           checkColor: GraphEditor.colorCurvature,
           side: BorderSide(color: GraphEditor.colorCurvature, width: 2),
           onChanged: (val) {
-            bool state =
-                widget.prefs.getBool(GraphEditor.prefShowCurvature) ?? true;
-            widget.prefs.setBool(GraphEditor.prefShowCurvature, !state);
-            widget.onUpdate();
+            widget.prefs.setBool(GraphEditor.prefShowCurvature, val!);
+            setState(() {
+              _showCurvature = val;
+              widget.onShouldRedraw();
+            });
```

Comment:
Being a bit pedantic here but I'd move the onShouldRedraw call out of setState to match all the other calls lol

```suggestion
            setState(() {
              _showCurvature = val;
            });
            widget.onShouldRedraw();
```

Reply:
Oops... <looks sheepishly away>